statchatui/src/Services/fetchData.js

```diff
@@ -0,0 +1,25 @@
+import { fetchStats } from "../Services/api";
+
+export async function fetchData(
+  isSignedIn,
+  setSignedInUserData,
+  onStatsChange
+) {
+  try {
+    if (isSignedIn) {
+      const profileResponse = await fetch(
+        `http://localhost:3002/users/profile`,
+        {
+          method: "GET",
+          credentials: "include",
+        }
+      );
+      const profileData = await profileResponse.json();
+      setSignedInUserData(profileData);
+      const stats = await fetchStats(profileData);
+      onStatsChange(stats);
```

Comment:
fetchData is still doing a lot of UI-specific stuff. Let's say you wanted to take this function and use it in an app that doesn't use React at all. Would it still make sense? Probably not! Let's remove **all** the parameters to this function.